guides/review.rst

```diff
@@ -224,9 +224,10 @@ New settings should:
    have more than one value for a setting across any of its runs, it should be
    some sort of global configuration, not a setting.
 
-Removing settings is not something we have done so far, so the exact process
-is still up in the air, but it should involve a careful deprecation path where
-the default behaviour does not change without first introducing warnings.
+When deprecating a setting for later removal, we prefer to change the default
+value of the setting to a private singleton (``not_set``), and implement the
+future behaviour immediately.  Passing any other value triggers a deprecation
+warning, but is otherwise a no-op (i.e. we still use the future behaviour).
```

Comment:
Hmm. I'm not sure we do this as reliably as this is suggesting. I think we have two distinct strategies:

* What you say here
* Keep the default the same but add a warning when the behaviour is triggered and add a special value that people can use to opt in to the new behaviour. This is e.g. what we did for `timeout` and is what I would imagine we would do if we removed the `buffer_size` setting (which I would like to do at some point)